openforcefield/tests/test_forcefield.py

```diff
@@ -1112,6 +1179,161 @@ def test_freesolv_parameters_assignment(self, freesolv_id, forcefield_version, a
+        pmd_struct.save(prmtop_file.name, overwrite=True)
+        pmd_struct.save(inpcrd_file.name, overwrite=True)
+
+        openmm_gbsas = {'HCT': openmm.app.HCT,
+                        'OBC1': openmm.app.OBC1,
+                        'OBC2': openmm.app.OBC2,
+                        }
+
+        # The functional form of the nonbonded force will change depending on whether the cutoff
+        # is None during initialization. Therefore, we need to figure that out here.
+
+        # WARNING: The NonbondedMethod enums at openmm.app.forcefield and openmm.CustomGBForce
+        # aren't necessarily the same, and could be misinterpreted if the wrong one is used. For
+        # create_system_from_amber, we must provide the app.forcefield version.
+
+        if is_periodic:
+            amber_nb_method = openmm.app.forcefield.CutoffPeriodic
+            amber_cutoff = off_nonbonded_force.getCutoffDistance()
+        else:
+            amber_nb_method = openmm.app.forcefield.NoCutoff
+            amber_cutoff = None
+
+        (amber_omm_system,
+         amber_omm_topology,
+         amber_positions) = create_system_from_amber(prmtop_file.name,
+                                                     inpcrd_file.name,
+                                                     implicitSolvent=openmm_gbsas[gbsa_model],
+                                                     nonbondedMethod = amber_nb_method,
+                                                     nonbondedCutoff=amber_cutoff,
+                                                     gbsaModel='ACE',
+                                                     implicitSolventKappa=0.,
+                                                     )
+
+        off_gbsa_force = [force for force in off_omm_system.getForces() if
+                              (isinstance(force, openmm.GBSAOBCForce) or
+                               isinstance(force, openmm.openmm.CustomGBForce))][0]
+        amber_gbsa_force = [force for force in amber_omm_system.getForces() if
+                                (isinstance(force, openmm.GBSAOBCForce) or
+                                 isinstance(force, openmm.openmm.CustomGBForce))][0]
+
+        # We get radius and screen values from each model's getStandardParameters method
+        if gbsa_model == 'HCT':
+            gb_params = openmm.app.internal.customgbforces.GBSAHCTForce.getStandardParameters(omm_top)
+        elif gbsa_model == 'OBC1':
+            gb_params = openmm.app.internal.customgbforces.GBSAOBC1Force.getStandardParameters(omm_top)
+        elif gbsa_model == 'OBC2':
+            gb_params = openmm.app.internal.customgbforces.GBSAOBC2Force.getStandardParameters(omm_top)
+
+        # Use GB params from OpenMM GBSA classes to populate parameters
+        for idx, (radius, screen) in enumerate(gb_params):
+            q, _, _2 = amber_gbsa_force.getParticleParameters(idx)
+            print(q, radius, screen, _, _2)
```

Comment:
what's the _, _2?

Reply:
i guess those are the amber radius/screens?

Reply:
That's right. And the print statement is some debugging code that I'll remove before the merge.

Reply:
7ae17c1 removes the print statement and adds a comment+more explicit variable names